### ISSUE TITLE
Wav2Vec 2.0 training model on completely different language need some help with parameters and info

### ISSUE THREAD

**Issue by amant555** [pytorch/fairseq#2557]
Hi,
I am training an ASR model on a completely different language from english using Wav2vec 2.0. Just need to know upto which  value of loss/accuracy I should train the model before moving to fine-tuning. 

Can you share info like:
1. Number of epochs trained for pre-training before moving to fine-tuning. 
2. Loss/accuracy required before moving on to fine-tuning.

Thanks in advance

**Comment by shamoons** [comment]
Will goo be using a pretrained model as a baseline or are you starting entirely from scratch?

**Comment by amant555** [comment]
I am starting from scratch.

**Comment by shamoons** [comment]
Why not start from a baseline pretrained model? I assume the vector representations of the audio will be similar enough across languages. You can then use those representations for your downstream ASR task. At least that’s how I think it’s supposed to work.

**Comment by amant555** [comment]
That's a good point. The reason I am training from scratch is because I want to know if training a model from scratch or using the baseline model yield any result change. I am also training a model on top of baseline model in parallel but there is stagnation for quite a while. And I am not sure if for a completely different language it will learn the correct quantized latent audio representation. So you can say that I am running an experiment here for my learning.

**Comment by shamoons** [comment]
Oh wow, that's pretty cool. I'd love to hear more about how your training from scratch works out. Please keep us posted

**Comment by luweishuang** [comment]
I finetune from english model "wav2vec_vox.pt", using 400+ hours chinese unsupervised audios. running until checkpoint_best.pt keep the same. It ran 17 epochs. Then I finetuned wav2vec_ctc model on chinese supervised audios, 200+ hours and running 14 epochs, finally I got a similarity cer/ler result with wav2letter "streaming_convnets" model which using 1000+ hours audios to training. The 200+ supervised dataset got from 1000+ hours proportionally and I used "LexiconFreeDecoder" in wav2letter and in fairseq's binding/python module.

**Comment by amant555** [comment]
Hi @luweishuang can you let me know what were the loss and accuracy value you got in pre-training( i.e before starting fine tuning). That would really help.

**Comment by luweishuang** [comment]
@amant555 the pretraining log maybe was deleted by me, sorry

**Comment by huihuifan** [comment]
This question is difficult to answer, as it depends on your dataset. I would train the model until it has converged and make sure it is not overfitting, before exploring finetuning.

**Comment by Leeyouxie** [comment]
> This question is difficult to answer, as it depends on your dataset. I would train the model until it has converged and make sure it is not overfitting, before exploring finetuning.

Hi, I'm trying to train a model from scratch using Arabic audio, but I found it is overfitting. The training set is about 500 hours.

**Comment by KadoshYa** [comment]
Hey I'm trying to train a model with  baseline pretrained model for Amharic language. I'm just starting up.

**Comment by sangam-kushwaha** [comment]
Hi everyone, I am a B.Tech final year student and I am trying to run  and train the Wav2Vec2.0 model from scratch  and I am getting error. Can someone please tell me steps in order to train the model from scratch that I should follow? That would really help me.
